lms/djangoapps/student_account/views.py

```diff
@@ -587,3 +591,38 @@ def account_settings_context(request):
         } for state in auth_states if state.provider.display_for_login or state.has_account]
 
     return context
+
+
+class RecoverPasswordView(ViewSet):
+    """Resets a password of a user by proving the email address
+
+        Example Requests:
+            POST /account/recover-password
+            {"email": "staff@example.com"}
+
+        Response Values:
+            HttpResponse: 200 if the password was reset correctly.
+            HttpResponse: 400 if email wasn't provided.
+            HttpResponse: 404 if user was not found with provided email.
+    """
+
+    def post(self, request, format=None):
+        email = request.data.get('email')
```

Comment:
The logic here is the same as in here: https://github.com/edx/edx-platform/blob/0a726a467924309845eb0ae6ebe0ed19f6d6d440/lms/djangoapps/student_account/views.py#L173-L173

Please refactor this code and the code above so there is a single path to the password reset logic.

Reply:
@macdiesel done